shell/platform/android/io/flutter/embedding/engine/dart/DartMessenger.java

```diff
@@ -33,22 +35,41 @@
 
   @NonNull private final FlutterJNI flutterJNI;
 
-  @NonNull private final ConcurrentHashMap<String, HandlerInfo> messageHandlers;
+  /**
+   * Maps a channel name to an object that contains the task queue and the handler associated with
+   * the channel.
+   *
+   * <p>Reads and writes to this map must lock {@code handlersLock}.
+   */
+  @NonNull
+  private final ConcurrentHashMap<String, HandlerInfo> messageHandlers = new ConcurrentHashMap<>();
```

Comment:
This no longer needs to be a `ConcurrentHashMap` now that all accesses are guarded by a lock.